### PR TITLE
Move location of gcloud gcs to storage

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Moved location of gcloud gcs to storage

The files which are under gcs directory, I kept them in [gcs directory](https://github.com/GoogleCloudPlatform/gcsfuse/blob/34c4fcf3d464b8a1a69bd4fd052b21b04f96a865/internal/storage/gcs/bucket.go) only as If I was putting it in to storage directory It was creating cycle between storage and storagutil. 

1. Moved [gcsutil](https://github.com/jacobsa/gcloud/tree/master/gcs/gcsutil) to bucketutil.
2. all the bucekts related files in bucket directory
bucket.go, bucket_handle.go, bucket_handle_test.go, storage_handle.go, storage_handle_test.go, debug_bucket.go, mock_bucket.go, fake_storage_util.go.
3. Moved Object related files in object directory.
Merge [gcloud/gcs/object.go](https://github.com/jacobsa/gcloud/blob/master/gcs/object.go) and [storage/object.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/storage/object.go)
object.go, object_test.go, object_attr.go,object_attr_test.go, request.go

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
3. Unit tests - NA
5. Integration tests - NA

### CODE REVIEW

**Comment by raj-prince** [comment]
@Tulsishah We need to raise PR directly on `tmp_sept_release` branch instead of `sept_release`.
CC: @vadlakondaswetha @sethiay

File: internal/storage/bucket_handle.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"net/http"
 
 	"cloud.google.com/go/storage"
-	"github.com/googlecloudplatform/gcsfuse/internal/gcloud/gcs"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
```
Comment (sethiay) [inline internal/storage/bucket_handle.go:29]: the files under storage/gcs can directly go under storage.

Comment (Tulsishah) [reply internal/storage/bucket_handle.go:29]: As I mentioned in the description
```
The files which are under gcs directory, I kept them in [gcs directory](https://github.com/GoogleCloudPlatform/gcsfuse/blob/34c4fcf3d464b8a1a69bd4fd052b21b04f96a865/internal/storage/gcs/bucket.go) only as If I was putting it in to storage directory It was creating cycle between storage and storagutil. 
```

Comment (sethiay) [reply internal/storage/bucket_handle.go:29]: Got it, i missed that. In that case, I would suggest to create `object` and 'bucket' packages under storage/. So the hierarchy would like:
internal/storage/
    object/
        object.go (copied from gcloud/gcs)
        ... (other object related files copied from gcloud/gcs and storage/ if any)
    bucket/
         object.go (copied from gcloud/gcs)
        ... (other bucket related files copied from gcloud/gcs and storage/ if any)    
    
    bucket_handle.go
    storage_handle.go
    requests.go 
    ...
    
CC' @vadlakondaswetha Does it look good to you ?

Comment (Tulsishah) [reply internal/storage/bucket_handle.go]: There are lot's of cycle forming due to this, I did some formation let me konw if it's correct.

Comment (sethiay) [reply internal/storage/bucket_handle.go]: Can you please give example ?

Is it possible to keep the following?
- storage/
  - object/
    - object.go
    - object_test.go
    - object_attrs.go
    - object_attrs_test.go
  - bucket/
    - bucket.go
    - debug_bucket.go
    - mock_bucket.go
  - requests/
    - request.go
  - storageutil/
    - Move everything from gcsutil to here.
  - errors.go
  - bucket_handle.go
  - bucket_handle_test.go
  - storage_handle.go
  - storage_handle_test.go
  - fake_storage_util.go
  - all other files

Comment (Tulsishah) [reply internal/storage/bucket_handle.go]: Object and request are forming cycle. Both should be in same directory.

Comment (sethiay) [reply internal/storage/bucket_handle.go]: In that case, object_attrs.go & object_attrs_test.go should be kept under storageutil/. Do you see any other issue with that ?

**Comment by Tulsishah** [comment]
Raised PR https://github.com/GoogleCloudPlatform/gcsfuse/pull/1334

**Pull request closed without merging**